File: content-delivery/src/models/WorkoutPlan.js
const mongoose = require('mongoose');

const WorkoutPlanSchema = new mongoose.Schema(
    {
        trainerId: {
            type: String,
            required: true
        },
        title: {
            type: String,
            required: true,
        },
        description: {
            type: String,
            required: true,
        },
        typeOfWorkout: {
            type: String,
            required: true,
        },
        videoIds: {
            type: String,
            required: true
        },
        likes: {
            type: [String],
            default: []
        },
        dislikes: {
            type: [String],
            default: []
        }
    },
    { timestamps: true }
);

module.exports = mongoose.model("WorkoutPlan", WorkoutPlanSchema);

File: dashboard/src/main/java/com/dashboard/user/WorkoutPlan.java
package com.dashboard.user;

import java.util.List;

import org.bson.types.ObjectId;
import org.springframework.data.mongodb.core.mapping.Document;

import jakarta.persistence.Id;

@Document
public class WorkoutPlan {

    @Id
    private ObjectId id;

    private String trainerId;

    private String title;

    private String description;

    private String typeOfWorkout;

    private List<String> videoIds;

    private List<String> likes;

    private List<String> dislikes;

}
